
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Documentation for the Progenetix oncogenomics resource">
      
      
        <meta name="author" content="Michael Baudis">
      
      
      
      
      
        
      
      
      <link rel="icon" href="../../img/site-logo-topright.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.1">
    
    
      
        <title>Classifications, Ontologies and Standards - Progenetix Documentation</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.484c7ddc.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.ab4e12ef.min.css">
      
      
  
  
    
      
      
    
  
  
  <style>:root{--md-admonition-icon--example:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2024%2024%22%3E%3Cpath%20d%3D%22M3.604%203.089A.75.75%200%200%201%204%203.75V8.5h.75a.75.75%200%200%201%200%201.5h-3a.75.75%200%201%201%200-1.5h.75V5.151l-.334.223a.75.75%200%200%201-.832-1.248l1.5-1a.75.75%200%200%201%20.77-.037M8.75%205.5a.75.75%200%200%200%200%201.5h11.5a.75.75%200%200%200%200-1.5zm0%206a.75.75%200%200%200%200%201.5h11.5a.75.75%200%200%200%200-1.5zm0%206a.75.75%200%200%200%200%201.5h11.5a.75.75%200%200%200%200-1.5zM5.5%2015.75c0-.704-.271-1.286-.72-1.686a2.3%202.3%200%200%200-1.53-.564c-.535%200-1.094.178-1.53.565-.449.399-.72.982-.72%201.685a.75.75%200%200%200%201.5%200c0-.296.104-.464.217-.564A.8.8%200%200%201%203.25%2015c.215%200%20.406.072.533.185.113.101.217.268.217.565%200%20.332-.069.48-.21.657-.092.113-.216.24-.403.419l-.147.14c-.152.144-.33.313-.52.504l-1.5%201.5a.75.75%200%200%200-.22.53v.25c0%20.414.336.75.75.75H5A.75.75%200%200%200%205%2019H3.31l.47-.47c.176-.176.333-.324.48-.465l.165-.156a6%206%200%200%200%20.536-.566c.358-.447.539-.925.539-1.593%22/%3E%3C/svg%3E');}</style>



    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../css/theme_overrides.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="deep-orange" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#list-of-filters-recognized-by-different-query-endpoints" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Progenetix Documentation" class="md-header__button md-logo" aria-label="Progenetix Documentation" data-md-component="logo">
      
  <img src="../../img/site-logo-topright.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Progenetix Documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Classifications, Ontologies and Standards
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/progenetix/progenetix-web" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M202.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1M496 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2m-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3m-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1"/></svg>
  </div>
  <div class="md-source__repository">
    progenetix-web
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Progenetix Documentation" class="md-nav__button md-logo" aria-label="Progenetix Documentation" data-md-component="logo">
      
  <img src="../../img/site-logo-topright.png" alt="logo">

    </a>
    Progenetix Documentation
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/progenetix/progenetix-web" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M202.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1M496 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2m-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3m-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1"/></svg>
  </div>
  <div class="md-source__repository">
    progenetix-web
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Documentation Home
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../news" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    News
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../changelog" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Data Changes
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../pgxRpi" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Analysis in R - pgxRpi
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../use-cases" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Use Case Examples
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../beacon-api" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Beacon+ API Use
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../services" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Services API
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../file-formats" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    File Formats
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../plotting" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Plotting & Visualizations
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Classifications, Ontologies & Standards
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../publication-collection" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Publication Collection
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../progenetix-data-review" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Data Review
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../technical-notes" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Technical Notes
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="http://progenetix.org" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Progenetix Data &#8599;
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="http://cancercelllines.org" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    cancercelllines.org &#8599;
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="http://baudisgroup.org" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Baudisgroup @ UZH &#8599;
  

    
  </span>
  
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#list-of-filters-recognized-by-different-query-endpoints" class="md-nav__link">
    <span class="md-ellipsis">
      
        List of filters recognized by different query endpoints
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="List of filters recognized by different query endpoints">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#public-ontologies-with-curie-based-syntax" class="md-nav__link">
    <span class="md-ellipsis">
      
        Public Ontologies with CURIE-based syntax
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#private-filters" class="md-nav__link">
    <span class="md-ellipsis">
      
        Private filters
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#diagnoses-phenotypes-and-histologies" class="md-nav__link">
    <span class="md-ellipsis">
      
        Diagnoses, Phenotypes and Histologies
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Diagnoses, Phenotypes and Histologies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ncit-coding-of-tumor-samples" class="md-nav__link">
    <span class="md-ellipsis">
      
        NCIt coding of tumor samples
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#icd-coding-of-tumor-samples" class="md-nav__link">
    <span class="md-ellipsis">
      
        ICD coding of tumor samples
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#uberon-codes" class="md-nav__link">
    <span class="md-ellipsis">
      
        UBERON codes
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#genomic-variations-cnv-ontology" class="md-nav__link">
    <span class="md-ellipsis">
      
        Genomic Variations (CNV Ontology)
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sequence-variation-snv-ontology" class="md-nav__link">
    <span class="md-ellipsis">
      
        Sequence Variation (SNV Ontology)
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-schemas" class="md-nav__link">
    <span class="md-ellipsis">
      
        Variant Schemas
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Variant Schemas">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ga4gh-variant-representation-schema" class="md-nav__link">
    <span class="md-ellipsis">
      
        GA4GH "Variant Representation" schema
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#links" class="md-nav__link">
    <span class="md-ellipsis">
      
        Links
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ad-hoc-community-formats" class="md-nav__link">
    <span class="md-ellipsis">
      
        Ad-Hoc &amp; "Community" Formats
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#progenetix-variant-schema" class="md-nav__link">
    <span class="md-ellipsis">
      
        Progenetix Variant schema
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#links_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Links
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#geolocation-data" class="md-nav__link">
    <span class="md-ellipsis">
      
        Geolocation Data
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Geolocation Data">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#provenance-and-use-of-geolocation-data" class="md-nav__link">
    <span class="md-ellipsis">
      
        Provenance and use of geolocation data
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="classifications-ontologies-and-standards">Classifications, Ontologies and Standards<a class="headerlink" href="#classifications-ontologies-and-standards" title="Permanent link">&para;</a></h1>
<p>The <a href="https://progenetix.org">Progenetix</a> resource utilizes standardized diagnostic coding
systems, with a dedicated move towards hierarchical ontologies. As part of the
coding process we have developed and provide several code mapping resources through repositories, the Progenetix website and APIs.</p>
<p>Additionally to diagnostic and other clinical concepts, <a href="https://progenetix.org">Progenetix</a>
increasingly uses hierarchical terms and concepts for the annotation and querying
of technical parameters such as platform technologies. Overall, the
<a href="https://progenetix.org">Progenetix</a> resource uses a query style based around the
<a href="https://beacon-project.io/v2/filters.html">Beacon v2 "filters"</a> concept with a
<a href="https://www.w3.org/TR/2010/NOTE-curie-20101216/">CURIE</a> based syntax.</p>
<hr />
<h2 id="list-of-filters-recognized-by-different-query-endpoints">List of filters recognized by different query endpoints<a class="headerlink" href="#list-of-filters-recognized-by-different-query-endpoints" title="Permanent link">&para;</a></h2>
<h3 id="public-ontologies-with-curie-based-syntax">Public Ontologies with CURIE-based syntax<a class="headerlink" href="#public-ontologies-with-curie-based-syntax" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>CURIE prefix</th>
<th>Code/Ontology</th>
<th>Examples</th>
</tr>
</thead>
<tbody>
<tr>
<td>NCIT</td>
<td>NCIt Neoplasm<sup id="fnref:1"><a class="footnote-ref" href="#fn:1">1</a></sup></td>
<td>NCIT:C27676</td>
</tr>
<tr>
<td>HP</td>
<td>HPO<sup id="fnref:2"><a class="footnote-ref" href="#fn:2">2</a></sup></td>
<td>HP:0012209</td>
</tr>
<tr>
<td>pubmed</td>
<td>NCBI Pubmed ID</td>
<td><a href="http://progenetix.org/services/ids/pubmed:18810378">pubmed:18810378</a></td>
</tr>
<tr>
<td>geo</td>
<td>NCBI Gene Expression Omnibus<sup id="fnref:3"><a class="footnote-ref" href="#fn:3">3</a></sup></td>
<td><a href="http://progenetix.org/services/ids/geo:GPL6801">geo:GPL6801</a>, <a href="http://progenetix.org/services/ids/geo:GSE19399">geo:GSE19399</a>, <a href="http://progenetix.org/services/ids/geo:GSM491153">geo:GSM491153</a></td>
</tr>
<tr>
<td>arrayexpress</td>
<td>EBI ArrayExpress<sup id="fnref:4"><a class="footnote-ref" href="#fn:4">4</a></sup></td>
<td>arrayexpress:E-MEXP-1008</td>
</tr>
<tr>
<td>cellosaurus</td>
<td>Cellosaurus - a knowledge resource on cell lines <sup id="fnref:5"><a class="footnote-ref" href="#fn:5">5</a></sup></td>
<td>cellosaurus:CVCL_1650</td>
</tr>
<tr>
<td>UBERON</td>
<td>Uberon Anatomical Ontology<sup id="fnref:6"><a class="footnote-ref" href="#fn:6">6</a></sup></td>
<td>UBERON:0000992</td>
</tr>
<tr>
<td>cbioportal</td>
<td>cBioPortal<sup id="fnref:9"><a class="footnote-ref" href="#fn:9">9</a></sup></td>
<td><a href="http://progenetix.org/services/ids/cbioportal:msk_impact_2017">cbioportal:msk_impact_2017</a></td>
</tr>
<tr>
<td>SO</td>
<td>Sequence Ontology<sup id="fnref:11"><a class="footnote-ref" href="#fn:11">11</a></sup></td>
<td><a href="http://progenetix.org/services/ids/SO:0000704">SO:0000704</a></td>
</tr>
</tbody>
</table>
<h3 id="private-filters">Private filters<a class="headerlink" href="#private-filters" title="Permanent link">&para;</a></h3>
<p>Since some classifications cannot directly be referenced, and in accordance with
the upcoming Beacon v2 concept of "private filters", <a href="https://progenetix.org">Progenetix</a> uses
additionally a set of structured non-CURIE identifiers.</p>
<p>For terms with a <code>pgx</code> prefix, the <a href="http://identifiers.org/pgx/">identifiers.org resolver</a> will </p>
<table>
<thead>
<tr>
<th>Filter prefix / local part</th>
<th>Code/Ontology</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr>
<td>pgx:icdom-...</td>
<td>ICD-O 3<sup id="fnref:7"><a class="footnote-ref" href="#fn:7">7</a></sup> Morphologies (Progenetix)</td>
<td>pgx:icdom-81703</td>
</tr>
<tr>
<td>pgx:icdot...</td>
<td>ICD-O 3<sup id="fnref2:7"><a class="footnote-ref" href="#fn:7">7</a></sup> Topographies(Progenetix)</td>
<td>pgx:icdot-C04.9</td>
</tr>
<tr>
<td>TCGA</td>
<td>The Cancer Genome Atlas (Progenetix)<sup id="fnref:8"><a class="footnote-ref" href="#fn:8">8</a></sup></td>
<td>TCGA-000002fc-53a0-420e-b2aa-a40a358bba37</td>
</tr>
<tr>
<td>pgx:cohort-...</td>
<td>Progenetix cohorts <sup id="fnref:10"><a class="footnote-ref" href="#fn:10">10</a></sup></td>
<td>pgx:cohort-arraymap</td>
</tr>
</tbody>
</table>
<hr />
<h2 id="diagnoses-phenotypes-and-histologies">Diagnoses, Phenotypes and Histologies<a class="headerlink" href="#diagnoses-phenotypes-and-histologies" title="Permanent link">&para;</a></h2>
<h3 id="ncit-coding-of-tumor-samples">NCIt coding of tumor samples<a class="headerlink" href="#ncit-coding-of-tumor-samples" title="Permanent link">&para;</a></h3>
<ul>
<li>based on <a href="https://evs.nci.nih.gov/ftp1/NCI_Thesaurus/Neoplasm/About_Core.html">NCIt neoplasm core</a>
but now extended based on the whole "neoplasia" subtree of the NCI Thesaurus (<code>NCIT:C3262</code> and child terms)</li>
<li>first implementation of NCIt concepts mapping in January 2017, then for a subset of arrayMap samples</li>
<li>now providing ICD-O 3 &lt;=&gt; NCIt mappings through the <a href="https://github.com/progenetix/ICDOntologies">ICDOntoologies</a> mapping project with a <a href="http://progenetix.org/service-collection/ontologymaps/">front-end an API</a> on the website</li>
</ul>
<h4 id="current-ncit-sample-codes">Current NCIt sample codes<a class="headerlink" href="#current-ncit-sample-codes" title="Permanent link">&para;</a></h4>
<ul>
<li><a href="http://progenetix.org/services/collations?collationTypes=NCIT&amp;deliveryKeys=id,label,count">progenetix.org/services/collations?collationTypes=NCIT&amp;deliveryKeys=id,label,count</a></li>
</ul>
<h3 id="icd-coding-of-tumor-samples">ICD coding of tumor samples<a class="headerlink" href="#icd-coding-of-tumor-samples" title="Permanent link">&para;</a></h3>
<p>The <a href="https://progenetix.org">Progenetix</a> resource primarily used the coding schemas of the _International Classification of Diseases in Oncology__ (3rd edition; "ICD-O 3"), to classify all biosamples for which experimental data is available. Users can get a list of ICD-O 3 codes in the Progenetix format <a href="http://info.progenetix.org/doc/services/collations.html">through Progenetix collations</a>.</p>
<p>The mappings used here for the ICD morphology codings (mapped to ICDMORPHOLOGY and ICDMORPHOLOGYCODE) are derived from the original source file last accessed on 2016-08-18 from <a href="https://www.who.int/standards/classifications/other-classifications/international-classification-of-diseases-for-oncology">the WHO</a>. The primary codes have been updated from the 2011 update document <a href="http://www.who.int/classifications/icd/updates/ICDO3Updates2011.pdf">ICDO3Updates2011.pdf</a>.</p>
<h4 id="current-icd-o-sample-codes">Current ICD-O sample codes<a class="headerlink" href="#current-icd-o-sample-codes" title="Permanent link">&para;</a></h4>
<ul>
<li>ICD-O Morphologies<ul>
<li><a href="http://progenetix.org/services/collations/?collationTypes=icdom&amp;deliveryKeys=id,label,count">progenetix.org/services/collations?collationTypes=icdom&amp;deliveryKeys=id,label,count</a></li>
</ul>
</li>
<li>ICD-O Topographies<ul>
<li><a href="http://progenetix.org/services/collations/?collationTypes=icdot&amp;deliveryKeys=id,label,count">progenetix.org/services/collations/?collationTypes=icdot&amp;deliveryKeys=id,label,count</a></li>
</ul>
</li>
</ul>
<h3 id="uberon-codes">UBERON codes<a class="headerlink" href="#uberon-codes" title="Permanent link">&para;</a></h3>
<p>The organ sites of the original coding have been mapped to UBERON. The mappings
are detailed in the related <a href="https://github.com/progenetix/icdot2uberon">icdot2uberon</a> project.</p>
<h4 id="current-uberon-sample-codes">Current UBERON sample codes<a class="headerlink" href="#current-uberon-sample-codes" title="Permanent link">&para;</a></h4>
<ul>
<li><a href="http://progenetix.org/services/collations?collationTypes=UBERON&amp;deliveryKeys=id,label,count">progenetix.org/services/collations?collationTypes=UBERON&amp;deliveryKeys=id,label,count</a></li>
</ul>
<hr />
<h2 id="genomic-variations-cnv-ontology">Genomic Variations (CNV Ontology)<a class="headerlink" href="#genomic-variations-cnv-ontology" title="Permanent link">&para;</a></h2>
<p><img src="../img/form-structural-variant-type-selector.png" style="float: right; width: 201px; margin-top: -15px; margin-left: 10px;"/>The <a href="https://progenetix.org">Progenetix</a> repository
contains a large amount of genomic copy number variants. While we had limited CNV type
annotations to the "minimum information content" - i.e. using <code>DUP</code> and <code>DEL</code>
categories for indicating relative genomic copy number gains or losses,
respectively - from 2022 <a href="https://progenetix.org">Progenetix</a> has moved to a richer CNV
classification in line with "common use practices".</p>
<p>As part of the <a href="http://cnvar.org">ELIXIR h-CNV community</a> and contributors
to the GA4GH <a href="http://genomebeacons.org">Beacon project</a> and
<a href="http://vrs.org">Variant Representation Specification (VRS)</a> we have co-developed
a "CNV assessment ontology" which in January 2022 has been accepted into the
<a href="https://www.ebi.ac.uk/ols/ontologies/efo">Experimental Factor Ontology (EFO)</a>,
has been adopted by the VRS 1.3 standard (w/ slight changes) and is under discussion
at <a href="https://github.com/The-Sequence-Ontology/SO-Ontologies/issues/568">Sequence Ontology (SO)</a>.</p>
<p>In January 2022 we switched the internal representation of CNV states to EFO codes
and implemented the respective search functionality in the <code>bycon</code> package. Future
data updates will gradually add the more granular classes such as <code>EFO:0030073</code>
where they apply.</p>
<div class="highlight"><pre><span></span><code>id: EFO:0030063
label: copy number assessment
  |
  |-id: EFO:0030064
  | label: regional base ploidy
  |   |
  |   |-id: EFO:0030065
  |     label: copy-neutral loss of heterozygosity
  |
  |-id: EFO:0030066
    label: relative copy number variation
      |
      |-id: EFO:0030067
      | label: copy number loss
      |   |
      |   |-id: EFO:0030068
      |   | label: low-level copy number loss
      |   |
      |   |-id: EFO:0030069
      |     label: complete genomic deletion
      |
      |-id: EFO:0030070
        label: copy number gain
          |
          |-id: EFO:0030071
          | label: low-level copy number gain
          |
          |-id: EFO:0030072
             label: high-level copy number gain
             note: commonly but not consistently used for &gt;=5 copies on a bi-allelic genome region
              |
              |-id: EFO:0030073
                 label: focal genome amplification
                 note: &gt;-
                   commonly used for localized multi-copy genome amplification events where the
                   region does not extend &gt;3Mb (varying 1-5Mb) and may exist in a large number of
                   copies
</code></pre></div>
<div class="admonition important">
<p class="admonition-title">CNV terminology</p>
<p>Please see the variants annotation table at cnvar.org or in the <a href="http://docs.genomebeacons.org/variant-queries/#term-use-comparison">Beacon v2 documentation</a>.</p>
</div>
<h2 id="sequence-variation-snv-ontology">Sequence Variation (SNV Ontology)<a class="headerlink" href="#sequence-variation-snv-ontology" title="Permanent link">&para;</a></h2>
<div class="tabbed-set tabbed-alternate" data-tabs="1:2"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1">Used terms</label><label for="__tabbed_1_2">Extended tree</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="highlight"><pre><span></span><code>id: SO:0001059
label: sequence_alteration
  |
  |-id: SO:0000159
  | label: deletion
  |
  |-id: SO:0000667
  | label: insertion
  |
  |-id: SO:1000002
  | label: substitution
      |
      |-id: SO:0002007
      | label: MNV (multiple nucleotide variant)
      |
      |-id: SO:0001483
        label: SNV (single nucleotide variant)
</code></pre></div>
</div>
<div class="tabbed-block">
<div class="highlight"><pre><span></span><code>id: SO:0001059
label: sequence_alteration
  |
  |-id: SO:0001744
  | label: UPD (uniparental disomy)
  |
  |-id: SO:0000159
  | label: deletion
  |
  |-id: SO:1000032
  | label: delins
  |
  |-id: SO:0000667
  | label: insertion
  |
  |-id: SO:1000036
  | label: inversion
  |
  |-id: SO:0000248
  | label: sequence_length_alteration
  |   |
  |   |-id: SO:0001019
  |   | label: copy_number_variation
  |   |   |
  |   |   |-id: SO:0001742
  |   |   | label: copy_number_gain
  |   |   |
  |   |   |-id: SO:0001743
  |   |   | label: copy_number_loss
  |   |   |
  |   |   |-id: SO:0002210
  |   |     label: presence_absence_variation
  |   |
  |   |-id: SO:0002096
  |   | label: short_tandem_repeat_variation
  |   |
  |   |-id: SO:0000207
  |     label: simple_sequence_length_variation
  |
  |-id: SO:0001785
  | label: structural_alteration
  |   |
  |   |-id: SO:0001784
  |   | label: complex_structural_alteration
  |   |   |
  |   |   |-id: SO:0002062
  |   |     label: complex_chromosomal_rearrangement
  |   |
  |   |-id: SO:0001872
  |   | label: rearrangement_region
  |   |
  |   |-id: SO:0000199
  |     label: translocation
  |       |
  |       |-id: SO:1000044
  |         label: chromosomal_translocation
  |
  |-id: SO:1000002
  | label: substitution
      |
      |-id: SO:0002007
      | label: MNV (multiple nucleotide variant)
      |
      |-id: SO:0001483
        label: SNV (single nucleotide variant)
</code></pre></div>
</div>
</div>
</div>
<table>
<thead>
<tr>
<th><a href="http://www.sequenceontology.org/">Sequence Ontology</a></th>
<th>Definition</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0001059"><code>SO:0001059</code></a> sequence_alteration</td>
<td>A sequence alteration is a sequence feature whose extent is the deviation from another sequence.</td>
</tr>
<tr>
<td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0001483"><code>SO:0001483</code></a> SNV</td>
<td>SNVs are single nucleotide positions in genomic DNA at which different sequence alternatives exist.</td>
</tr>
<tr>
<td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0002007"><code>SO:0002007</code></a> MNV</td>
<td>An MNV is a multiple nucleotide variant (substitution) in which the inserted sequence is the same length as the replaced sequence.</td>
</tr>
<tr>
<td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0000159"><code>SO:0000159</code></a> deletion</td>
<td>The point at which one or more contiguous nucleotides were excised.</td>
</tr>
<tr>
<td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0000667"><code>SO:0000667</code></a> insertion</td>
<td>The sequence of one or more nucleotides added between two adjacent nucleotides in the sequence.</td>
</tr>
</tbody>
</table>
<h2 id="variant-schemas">Variant Schemas<a class="headerlink" href="#variant-schemas" title="Permanent link">&para;</a></h2>
<h3 id="ga4gh-variant-representation-schema">GA4GH "Variant Representation" schema<a class="headerlink" href="#ga4gh-variant-representation-schema" title="Permanent link">&para;</a></h3>
<p>The "Genomic Knowledge Standards" (GKS) of the Global Alliance for Genomics and
Health <a href="http://ga4gh.org">GA4GH</a> develops a modern schema for the unambiguous
representation, transmission and recovery of sequence variants (genomic and
beyond).
The first release of the GA4GH Variation Representation Specification
(<a href="https://github.com/ga4gh/vr-spec/releases/tag/1.0.0">vr-spec v1.0</a>)
does not yet include the option to represent structural variants. However, the
internal roadmap of the project points towards an extension for CNV
representation in 2020.</p>
<h3 id="links">Links<a class="headerlink" href="#links" title="Permanent link">&para;</a></h3>
<ul>
<li><em>vr-spec</em> <a href="https://github.com/ga4gh/vr-spec">repository</a></li>
<li><a href="https://vr-spec.readthedocs.io/en/1.0/">documentation</a></li>
</ul>
<h3 id="ad-hoc-community-formats">Ad-Hoc &amp; "Community" Formats<a class="headerlink" href="#ad-hoc-community-formats" title="Permanent link">&para;</a></h3>
<h3 id="progenetix-variant-schema"><em>Progenetix</em> <code>Variant</code> schema<a class="headerlink" href="#progenetix-variant-schema" title="Permanent link">&para;</a></h3>
<p>The <a href="http://progenetix.org">Progenetix</a> cancer genomics resource store their millions of CNVs
in as data objects in <a href="http://mongodb.org">MongoDB</a> document databases. The
format of the single variants is based on the Beacon v2 default model with some
modifications (e.g. incorporating the VRS 1.3 <code>RelativeCopyNumber</code> concept but
w/ slightly rewrapped components).</p>
<p>The <em>Progenetix</em> data serves as the repository behind the
<a href="http://beaconplus.progenetix.org">Beacon<span style="color: red; font-weight: 800;">+</span></a>
forward looking implementation of the <a href="https://genomebeacons.org">ELIXIR Beacon</a> project.
Accordingly, upon export through the API variants are re-mapped to a Beacon v2 representation.</p>
<h4 id="progenetix-cnv-example">Progenetix CNV example<a class="headerlink" href="#progenetix-cnv-example" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="p">{</span>
<span class="w">  </span><span class="nt">&quot;id&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;pgxvar-5bab576a727983b2e00b8d32&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="nt">&quot;variant_internal_id&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;11:52900000-134452384:DEL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="nt">&quot;analysis_id&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;pgxcs-kftvldsu&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="nt">&quot;biosample_id&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;pgxbs-kftva59y&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="nt">&quot;individual_id&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;pgxind-kftx25eh&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="nt">&quot;variant_state&quot;</span><span class="p">:</span><span class="w"> </span><span class="p">{</span><span class="w"> </span><span class="nt">&quot;id&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;EFO:0030067&quot;</span><span class="p">,</span><span class="w"> </span><span class="nt">&quot;label&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;copy number loss&quot;</span><span class="w"> </span><span class="p">},</span>
<span class="w">  </span><span class="nt">&quot;location&quot;</span><span class="p">:</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nt">&quot;sequence_id&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;refseq:NC_000011.10&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;chromosome&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;11&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;start&quot;</span><span class="p">:</span><span class="w"> </span><span class="mi">52900000</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;end&quot;</span><span class="p">:</span><span class="w"> </span><span class="mi">134452384</span>
<span class="w">    </span><span class="p">},</span>
<span class="w">  </span><span class="nt">&quot;updated&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;2022-03-29T14:36:47.454674&quot;</span>
<span class="p">}</span>
</code></pre></div>
<h4 id="progenetix-snv-example">Progenetix SNV example<a class="headerlink" href="#progenetix-snv-example" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="w">  </span><span class="p">{</span>
<span class="w">    </span><span class="nt">&quot;updated&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;2023-05-25T17:03:45.096849&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;analysis_id&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;pgxcs-kl8hg1r8&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;biosample_id&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;pgxbs-kl8hg1r4&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;id&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;pgxvar-5be1840772798347f0ed9d9d&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;variant_internal_id&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;5:67589139:G&gt;A&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;location&quot;</span><span class="p">:</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="nt">&quot;sequence_id&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;refseq:NC_000005.10&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="nt">&quot;chromosome&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;5&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="nt">&quot;start&quot;</span><span class="p">:</span><span class="w"> </span><span class="mi">67589138</span><span class="p">,</span>
<span class="w">      </span><span class="nt">&quot;end&quot;</span><span class="p">:</span><span class="w"> </span><span class="mi">67589139</span>
<span class="w">    </span><span class="p">},</span>
<span class="w">    </span><span class="nt">&quot;individual_id&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;pgxind-kl8hg1r5&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;reference_sequence&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;G&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;sequence&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;A&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;variant_state&quot;</span><span class="p">:</span><span class="w"> </span><span class="p">{</span><span class="w"> </span><span class="nt">&quot;id&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;SO:0001059&quot;</span><span class="p">,</span><span class="w"> </span><span class="nt">&quot;label&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;sequence_alteration&quot;</span><span class="w"> </span><span class="p">}</span>
<span class="w">  </span><span class="p">}</span>
</code></pre></div>
<h3 id="links_1">Links<a class="headerlink" href="#links_1" title="Permanent link">&para;</a></h3>
<ul>
<li>schema in <em>progenetix/bycon</em> <a href="https://github.com/progenetix/bycon/blob/master/schemas/src/progenetix-database-schemas/pgxVariant.yaml">code repository</a></li>
</ul>
<hr />
<h2 id="geolocation-data">Geolocation Data<a class="headerlink" href="#geolocation-data" title="Permanent link">&para;</a></h2>
<h3 id="provenance-and-use-of-geolocation-data">Provenance and use of geolocation data<a class="headerlink" href="#provenance-and-use-of-geolocation-data" title="Permanent link">&para;</a></h3>
<p>Geographic point coordinates are assigned to each sample after review of existing information from associated publications or repository information for their ”best available” geographic origin using a precedence of:  </p>
<ol>
<li>sample specific data (e.g. from article text)</li>
<li>experiment location</li>
<li>first author proxy</li>
</ol>
<p>For publications w/o accessible sample data in general the "author proxy" is being used, unless specific annotations have been found in the article.</p>
<p>A more detailed discussion of the problems and benefits of geographic provenance tagging can be found in <a href="https://academic.oup.com/database/article/doi/10.1093/database/baaa009/5812711">Carrio-Cordo <em>et al.</em>, DATABASE 2020</a>.</p>
<h4 id="geolocations-service">Geolocations Service<a class="headerlink" href="#geolocations-service" title="Permanent link">&para;</a></h4>
<p>The <a href="https://progenetix.org">Progenetix</a> API provides a service for <a href="/https://docs.progenetix.org/services/#geographic-locations-cities-geolocations">retrieving geographic coordinates</a> as point coordinates, for the majority of cities.</p>
<h4 id="geolocation-schema"><code>GeoLocation</code> schema<a class="headerlink" href="#geolocation-schema" title="Permanent link">&para;</a></h4>
<p>The current version of the JSON Schema data schema for the geolocation object can be accessed through the <a href="http://progenetix.org/services/schemas/GeoLocation/">Progenetix services API</a>.</p>
<div class="highlight"><pre><span></span><code>&quot;geometry&quot;: {
  &quot;coordinates&quot;: [
    8.69,
    49.41
  ],
  &quot;type&quot;: &quot;Point&quot;
},
&quot;properties&quot;: {
  &quot;ISO3166alpha2&quot;: &quot;DE&quot;,
  &quot;ISO3166alpha3&quot;: &quot;DEU&quot;,
  &quot;city&quot;: &quot;Heidelberg&quot;,
  &quot;continent&quot;: &quot;Europe&quot;,
  &quot;country&quot;: &quot;Germany&quot;
},
&quot;type&quot;: &quot;Feature&quot;
</code></pre></div>
<hr />
<div class="footnote">
<hr />
<ol>
<li id="fn:1">
<p>National Cancer Institute Thesaurus Neoplasm <a href="https://bioportal.bioontology.org/ontologies/NCIT_NEOPLASM">NCIt Neoplasm</a>&#160;<a class="footnote-backref" href="#fnref:1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="fn:2">
<p>Human phenotype ontology <a href="https://hpo.jax.org">HPO</a>&#160;<a class="footnote-backref" href="#fnref:2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
<li id="fn:3">
<p>Supported identifiers include platforms(GPL), series(GSE) and samples(GSM).<a href="https://www.ncbi.nlm.nih.gov/geo/info/overview.html">GEO Overview</a>&#160;<a class="footnote-backref" href="#fnref:3" title="Jump back to footnote 3 in the text">&#8617;</a></p>
</li>
<li id="fn:4">
<p>Supports ArrayExpress Accession ID. <a href="https://www.ebi.ac.uk/arrayexpress/browse.html">ArrayExpress browse</a>&#160;<a class="footnote-backref" href="#fnref:4" title="Jump back to footnote 4 in the text">&#8617;</a></p>
</li>
<li id="fn:5">
<p><a href="https://www.cellosaurus.org/">Cellosaurus</a> accession ID.&#160;<a class="footnote-backref" href="#fnref:5" title="Jump back to footnote 5 in the text">&#8617;</a></p>
</li>
<li id="fn:6">
<p><a href="http://uberon.github.io/about.html">Uberon</a> ID&#160;<a class="footnote-backref" href="#fnref:6" title="Jump back to footnote 6 in the text">&#8617;</a></p>
</li>
<li id="fn:7">
<p>International Classification of Diseases for Oncology, 3rd Edition <a href="https://www.who.int/standards/classifications/other-classifications/international-classification-of-diseases-for-oncology">ICD-O-3</a>&#160;<a class="footnote-backref" href="#fnref:7" title="Jump back to footnote 7 in the text">&#8617;</a><a class="footnote-backref" href="#fnref2:7" title="Jump back to footnote 7 in the text">&#8617;</a></p>
</li>
<li id="fn:8">
<p>Supports <a href="https://portal.gdc.cancer.gov">TCGA</a> Sample UUID.&#160;<a class="footnote-backref" href="#fnref:8" title="Jump back to footnote 8 in the text">&#8617;</a></p>
</li>
<li id="fn:9">
<p>Supports <a href="https://www.cbioportal.org/datasets">cBioPortal</a> Study ID.&#160;<a class="footnote-backref" href="#fnref:9" title="Jump back to footnote 9 in the text">&#8617;</a></p>
</li>
<li id="fn:10">
<p>Cohorts defined in Progenetix involving a collection of related samples. Currently includes (add <code>pgx:cohort-</code>): arraymap, 2021progenetix, DIPG, TCGA, TCGAcancers, <a href="http://progenetix.org/progenetix-cohorts/gao-2021-signatures/">gao2021signatures</a>.&#160;<a class="footnote-backref" href="#fnref:10" title="Jump back to footnote 10 in the text">&#8617;</a></p>
</li>
<li id="fn:11">
<p><a href="https://www.sequenceontology.org/">Sequence Ontology</a> ID&#160;<a class="footnote-backref" href="#fnref:11" title="Jump back to footnote 11 in the text">&#8617;</a></p>
</li>
</ol>
</div>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      &copy; Copyright 2023, Michael Baudis and progenetix.org
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
      
    
    
    
      
      
    
    <a href="https://genomic.social/@progenetix" target="_blank" rel="noopener me" title="genomic.social" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M433 179.1c0-97.2-63.7-125.7-63.7-125.7-62.5-28.7-228.6-28.4-290.5 0 0 0-63.7 28.5-63.7 125.7 0 115.7-6.6 259.4 105.6 289.1 40.5 10.7 75.3 13 103.3 11.4 50.8-2.8 79.3-18.1 79.3-18.1l-1.7-36.9s-36.3 11.4-77.1 10.1c-40.4-1.4-83-4.4-89.6-54-.6-4.6-.9-9.3-.9-13.9 85.6 20.9 158.7 9.1 178.7 6.7 56.1-6.7 105-41.3 111.2-72.9 9.8-49.8 9-121.5 9-121.5zm-75.1 125.2h-46.6V190.1c0-49.7-64-51.6-64 6.9v62.5H201V197c0-58.5-64-56.6-64-6.9v114.2H90.3c0-122.1-5.2-147.9 18.4-175 25.9-28.9 79.8-30.8 103.8 6.1l11.6 19.5 11.6-19.5c24.1-37.1 78.1-34.8 103.8-6.1 23.7 27.3 18.4 53 18.4 175"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://github.com/progenetix" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "../..", "features": ["content.tabs.link", "search.highlight", "search.share", "navigation.instant", "navigation.tracking", "navigation.sections"], "search": "../../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.79ae519e.min.js"></script>
      
    
  </body>
</html>