---
title: "Population assignment from cancer genomes"
date: 2020-03-16
template: post.html
authors:
  - "@mbaudis"
  - "@qingyao"
excerpt_link: https://baudisgroup.org/publications/2020-03-16-snp2pop-scientificreports/
pdf_file_name: 2020-03-16___Huang-Q-and-Baudis-M__Population-assignment-from cancer genomes-with-SNP2pop__ScientificReports.pdf
pdf_file_type: article    # slides poster article
www_link:
www_links_formatted:
  - final article at <a href="https://www.nature.com/articles/s41598-020-61854-x" target="_blank">[ScientificReports]</a>
  - earlier <a href="https://www.biorxiv.org/content/early/2019/01/14/368647" target="_blank">[bioRxiv]</a> version
category:
  - about
  - news
tags:
  - article
---

### Enabling population assignment from cancer genomes with SNP2pop.
#### Huang Q and Baudis M. (2020)
##### Sci Rep 10, 4846 (2020). <https://doi.org/10.1038/s41598-020-61854-x>

<!--more-->

**Abstract** In many cancers, incidence, treatment efficacy and overall prognosis vary between geographic populations. Studies disentangling the contributing factors may help in both understanding cancer biology and tailoring therapeutic interventions. Ancestry estimation in such studies should preferably be driven by genomic data, due to frequently missing or erroneous self-reported or inferred metadata. While respective algorithms have been demonstrated for baseline genomes, such a strategy has not been shown for cancer genomes carrying a substantial somatic mutation load. We have developed a bioinformatics tool for the assignment of population groups from genome profiling data for both unaltered and cancer genomes. Despite extensive somatic mutations in the cancer genomes, consistency between germline and cancer data reached of 97% and 92% for assignment into 5 and 26 ancestral groups, respectively. Comparison with self-reported meta-data estimated a matching rate between 88â€“92%, mostly limited by interpretation of self-reported ethnicity labels compared to the standardized mapping output. Our SNP2pop application allows to assess population information from SNP arrays as well as sequencing platforms and to estimate the population structure in cancer genomics projects, to facilitate research into the interplay between ethnicity-related genetic background, environmental factors and somatic mutation patterns in cancer biology.
